### PR TITLE
[ALLUXIO-818] Clean up orphaned blocks on register.

### PR DESCRIPTION
**Pull request by calvinjia**
https://alluxio.atlassian.net/browse/ALLUXIO-818

### CODE REVIEW

**Comment by calvinjia** [comment]
alluxio-bot, check this please.

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 1]
```diff
@@ -708,12 +709,24 @@ private void processWorkerAddedBlocks(MasterWorkerInfo workerInfo,
             mLostBlocks.remove(blockId);
           }
         } else {
-          LOG.warn("Failed to register workerId: {} to blockId: {}", workerInfo.getId(), blockId);
+          LOG.warn("Invalid block: {} from worker {}.", workerInfo.getWorkerAddress().getHost(),
```
Comment (apc999) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: wrong  arg order?

File: core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java [hunk 2]
```diff
@@ -199,6 +199,21 @@ public void removeBlockTellsWorkersToRemoveTheBlock() throws Exception {
     assertEquals(ImmutableList.of(1L), heartBeat.getData());
   }
 
+  @Test
+  public void registerCleansUpOrphanedBlocks() throws Exception {
+    // Create a worker with a unknown blocks.
```
Comment (apc999) [inline core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java]: nit: an unknown

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java (hunk 1 above)
Comment (calvinjia) [reply core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: Fixed, thanks.

File: core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java (hunk 2 above)
Comment (calvinjia) [reply core/server/master/src/test/java/alluxio/master/block/BlockMasterTest.java]: Fixed, thanks.

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 3]
```diff
@@ -708,12 +709,24 @@ private void processWorkerAddedBlocks(MasterWorkerInfo workerInfo,
...
+          LOG.warn("Invalid block: {} from worker {}.", blockId,
+              workerInfo.getWorkerAddress().getHost());
         }
       }
     }
   }
 
+  @GuardedBy("workerInfo")
+  private void processWorkerOrphanedBlocks(MasterWorkerInfo workerInfo) {
+    for (long block: workerInfo.getBlocks()) {
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: NIT: should there be a space before `:`?

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 4]
```diff
@@ -708,12 +709,24 @@ private void processWorkerAddedBlocks(MasterWorkerInfo workerInfo,
...
+      if (!mBlocks.containsKey(block)) {
+        LOG.warn("Requesting delete for orphaned block: {} from worker {}.", block,
```
Comment (aaudiber) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: I'd prefer `info` here; we already warn in `processWorkerAddedBlocks`, and the message sounds more informational than warn-like

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java (hunk 3 above)
Comment (calvinjia) [reply core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: Fixed, thanks.

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java (hunk 4 above)
Comment (calvinjia) [reply core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java]: Fixed, thanks.

**Pull request merged**